packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingTask.js

```diff
@@ -38,18 +38,15 @@ const Drawing = types.model('Drawing', {
     }
 
     function complete () {
-      const newValue = self.marks.map(mark => detach(mark))
-      self.updateAnnotation(newValue)
+      self.updateAnnotation(self.marks)
     }
 
     function start () {
       const activeMarks = self.annotation.value
-      activeMarks.forEach(function addMarksToTools (mark) {
-        const newMark = clone(mark)
-        const tool = tryReference(() => self.tools[newMark.toolIndex])
-        tool && tool.marks.put(newMark)
-      })
-      self.updateAnnotation([])
+      // if the current annotation is empty, clear any leftover marks.
+      if (activeMarks.length === 0) {
+        self.tools.forEach(tool => tool.marks.clear())
+      }
```

Comment:
Same comment about this change. A small explanation would be helpful since what this is doing is pretty different from before and I'm not sure how getting the mark's tool using MST's helper impacts this.

Reply:
This is clearing any stale marks from each tool if the current annotation is empty when the task starts. The classifier doesn't seem to reset tasks on new subjects/classifications (as far as I can tell.) I think #1365 would help here too.

Reply:
The `WorkflowStepStore` observes for the `onSubjectReady` action and resets itself which I think at one point managed the tasks and reset them, but some refactoring has happened since I wrote it, so I don't think it does that anymore. I think steps should be responsible for tasks and manage resetting itself rather than have this happen here. It's confusing to me.

Reply:
Making changes to the steps to manage clearing tasks is out of scope of this PR though.

Reply:
I found a bug in the subject/classification lifecycle, which should have been fixed in a recent PR (can't remember which one though.) New subjects rendered marks from the previous subject, because `onSubjectReady` was firing before the new classification was created.

Without docs, or examples, I do find it hard to follow what happens where in the classifier models. That's why I kept this code close to where it's needed. I'd be absolutely fine with new subjects calling something like `task.init()` or `task.reset()` on all workflow tasks before a new classification starts.

Perhaps rename `task.start()` to make it absolutely clear that it happens each time you navigate to that task during the current classification, rather than once when the new subject loads? The name seems ambiguous to me at the moment.

Reply:
Yeah, a rename on `task.start()` sounds good. Do you mind opening a couple of issues to track these items?

Reply:
https://github.com/zooniverse/front-end-monorepo/pull/1339 fixed the bug where each new subject starts before `classificationComplete` for the previous subject.

Reply:
Looking at the changes in #1339, the workflow steps store does reset each step task, by calling `task.start()`, when a new subject loads.